File: src/manifest.ts
export function getManifest() {
  return {
    id: "mini-media-server.addon",
    version: "1.0.0",
    name: "Mini Media Server",
    description:
      "A tiny Stremio addon acting similar to a self-hostable debrid service. Search for torrents via Jackett, download and cache them automatically and serve them to Stremio.",
    resources: ["stream"],
    types: ["movie", "series"],
    catalogs: [],
  };
}

File: src/routes/manifest.rs
use axum::{extract::{Path, State}, http::StatusCode, response::{IntoResponse, Json, Response}};
use serde_json::json;
use crate::routes::AppState;

pub async fn manifest(
    State(state): State<AppState>,
    Path(api_key): Path<String>,
) -> Result<impl IntoResponse, Response> {
    if !state.db.validate_api_key(&api_key) {
        return Err((StatusCode::UNAUTHORIZED, "Invalid API Key").into_response());
    }

    Ok(Json(json!({
        "id": "mini-media-server.addon",
        "version": "1.0.0",
        "name": "Mini Media Server",
        "description": "A tiny Stremio addon acting similar to a self-hostable debrid service.",
        "resources": ["stream"],
        "types": ["movie", "series"],
        "catalogs": []
    })))
}
